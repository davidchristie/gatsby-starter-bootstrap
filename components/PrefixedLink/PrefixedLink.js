import { prefixLink } from 'gatsby-helpers'
import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router'

class PrefixedLink extends React.Component {
  render () {
    const { className, to } = this.props
    return (
      <Link className={className} to={prefixLink(to)}>
        {this.props.children}
      </Link>
    )
  }
}

PrefixedLink.propTypes = {
  className: PropTypes.string,
  to: PropTypes.string
}

export default PrefixedLink
