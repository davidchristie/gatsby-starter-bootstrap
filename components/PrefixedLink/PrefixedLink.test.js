import React from 'react'
import renderer from 'react-test-renderer'

import PrefixedLink from './PrefixedLink'

it('renders with errors', () => {
  renderer.create(<PrefixedLink />)
})

it('matches render snapshot', () => {
  const tree = renderer.create(<PrefixedLink />).toJSON()
  expect(tree).toMatchSnapshot()
})
