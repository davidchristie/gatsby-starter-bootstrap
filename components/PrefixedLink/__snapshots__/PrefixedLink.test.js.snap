// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches render snapshot 1`] = `
<a
  className={undefined}
  onClick={[Function]}
  style={Object {}}
/>
`;
